esp/services/ws_workunits/ws_workunitsService.cpp

```diff
@@ -3354,11 +3354,30 @@ bool CWsWorkunitsEx::onWUProcessGraph(IEspContext &context,IEspWUProcessGraphReq
             throw MakeStringException(ECLWATCH_CANNOT_OPEN_WORKUNIT,"Cannot open workunit %s.",wuid.str());
         ensureWsWorkunitAccess(context, *cw, SecAccess_Read);
 
-        Owned <IConstWUGraph> graph = cw->getGraph(req.getName());
-        Owned <IPropertyTree> xgmml = graph->getXGMMLTree(true); // merge in graph progress information
+        unsigned count = cw->getGraphCount();
+        if (count < 1)
+            throw MakeStringException(ECLWATCH_GRAPH_NOT_FOUND, "No graph found in %s.",wuid.str());
+
+        if (isEmpty(req.getName()) && (count > 1))
```

Comment:
I thought that the getGraphCount() is not expensive. I will remove the code related to the count.